### DISCUSSION TITLE
Outputting all leaf keys (including duplicates) in a yaml file?

### DISCUSSION THREAD

**Discussion by embee1981** [mikefarah/yq#1139] [Q&A]
Hi all

Apologies for yet another question.

I'm looking to output all the leaf keys in a yaml file such that I can flag if there are any duplicates present.

For a yaml file like:
```
bob: hello alice
bob: hello alice
something: here
blah:
  test: hello
  test: goodbye
  blah: argh
  more:
    nesting: here
```

I want to get yq to output:
```
bob
bob
something
blah.test
blah.test
blah.blah
blah.more.nesting
```

This would let me do something like

`yq '<some command>' myFile.yaml | uniq -cd`
and see
```
2 bob
2 blah.test
```

Something like
`yq '(keys)[]'`
will only handle the top level things:
```
bob
bob
something
blah
```
Looking at another discussion question, I can get close with 
`yq '.. | select((tag == "!!map" or tag == "!!seq") | not) | (path | join("."))'` 
which will give me all the keys but not the dupes:
```
bob
something
blah.test
blah.blah
blah.more.nesting
```
and
`yq '... | select((tag == "!!map" or tag == "!!seq") | not) | (path | join("."))'`
which shows the dupes but is showing too much
```
bob
bob
something
something
blah
blah.test
blah.test
blah.blah
blah.blah
blah.more
blah.more.nesting
blah.more.nesting
```
Is this possible with yq to get all keys, regardless of duplication, to output?

**Comment by mikefarah** [comment]
It's not valid yaml to have duplicate key names in a map

**Reply by embee1981** [reply]
Yeah the spec seems to suggest its up to the parser as to what to do here. yq allows the dupes through but I'd just like to flag there is a problem, either via a query or yq blowing up when it encounters something like this (with a nice error message)
